README.md

```diff
@@ -76,6 +76,29 @@ And this is what gets rendered (plus the CSS and Javascript you've specified):
 
 ## Release notes
 
+🚨📢 **Version 0.110**
+- BREAKING CHANGE:
+    - Installation changes:
+        - If your components include JS or CSS, you now must use the middleware and add django-components' URLs to your `urlpatterns`
+            (See "[Adding support for JS and CSS](#adding-support-for-js-and-css)")
+    - `preload` kwarg of `{% component_dependencies %}` tag was removed. This will be replaced with HTML fragment support.
```

Comment:
The removal of the `preload` kwarg from `{% component_dependencies %}` tags can be [seen here](https://github.com/EmilStenstrom/django-components/pull/688/files#diff-7345171ed38e1dd2e4b8199b65a2fd082323bdb331ff0ef0dc0dd3bd43f111aeL58).

It had to be removed, because in the new format, when we render component and insert a `<!-- _RENDERED -->` comment, we also need to know the ID of the component instance.

And we don't have any ID when we just specify the name of the component.

---

Moreover, I believe the `preload` kwarg was likely a workaround for allowing to render components as HTML fragments - E.g. that one would list the names of the components that *may* be loaded into the page AFTER the HTML document has loaded.

And this would be addressed separately in https://github.com/EmilStenstrom/django-components/issues/635, with which the `preload` kwarg would be no longer needed anyway.

---

Taking the HTML fragment support out of the consideration, I can imagine that there could still be a desire for "pre-loading" JS / CSS.

The use case would be one of optimization, for example if:
- A web dev used HTML fragments approach.
- One of the fragments would depend on large JS / CSS files that would be loading slowly.
- HTML fragments are sent on end-user's actions. And so if, during user actions, user would have to wait for the JS / CSS to load, this would be bad user experience.
- In such case the web dev may want to pre-load the fragments' JS / CSS, so at the point of user's action, everything is already loaded.

However, to support pre-loading PROPERLY, we'd have to take into the account that there can be multiple instances of component registries, and that there may be components NOT registered to any `ComponentRegistry` (which the old `preload` kwarg didn't account for).

So, instead, the preloading should be specified on the component, to avoid passing around the reference to it. But it may still be that for one page a component SHOULD be preloaded, and for a different page, it should NOT.

So IMO the component should specify for which "conditions" it should be preloaded and then link that up in the `{% component_dependencies %}`.

So it could look like this:
```py
class MyComp(Component):
    # Component specifies for which "preload_ids"
    # it should have its CSS / JS preloaded
    preload = ["page_1", "page_2", "page_3"]
```

```django
{# And the "preload_id" on `component_dependencies` #}
{# would load only those components that have a matching ID #}
{% component_dependencies id="page_1" %}
```

@EmilStenstrom Let me know what you think of this. IMO it's low on priority, but I can make it into a standalone issue to see if people would want it.

Reply:
No harm in checking, so I made an issue for it here https://github.com/EmilStenstrom/django-components/issues/701